### PR TITLE
Managed IngressController Pre-Checks

### PR DESCRIPTION
**Pull request by dustman9000**
Protect from adding/deleting hive/cloud-ingress-operator managed ingresscontrollers (default, apps2). 
Add labels to managed resources.

### CODE REVIEW

File: pkg/controller/customdomain/customdomain_utils.go [hunk 1]
```diff
@@ -112,12 +112,16 @@ func (r *ReconcileCustomDomain) finalizeCustomDomain(reqLogger logr.Logger, inst
 	}, ingressSecret)
 	if err != nil {
 		reqLogger.Error(err, fmt.Sprintf("Failed to get %s secret", instance.Name))
-		return err
-	}
-	err = r.client.Delete(context.TODO(), ingressSecret)
-	if err != nil {
-		reqLogger.Error(err, fmt.Sprintf("Failed to delete %s secret", instance.Name))
-		return err
+	} else {
```
Comment (jharrington22) [inline pkg/controller/customdomain/customdomain_utils.go:115]: Should `err != nil` return an error and then remove this else block?

File: pkg/controller/customdomain/customdomain_utils.go [hunk 2]
```diff
@@ -128,15 +132,18 @@ func (r *ReconcileCustomDomain) finalizeCustomDomain(reqLogger logr.Logger, inst
 	}, customIngress)
 	if err != nil {
 		reqLogger.Error(err, fmt.Sprintf("Failed to get %s ingresscontroller", instance.Name))
-		return err
-	}
-	err = r.client.Delete(context.TODO(), customIngress)
-	if err != nil {
-		reqLogger.Error(err, fmt.Sprintf("Failed to delete %s ingresscontroller", instance.Name))
-		return err
+	} else {
```
Comment (jharrington22) [inline pkg/controller/customdomain/customdomain_utils.go:135]: Same here return an error if we fail to get the ingresscontroller to try again? Remove the else?

**Review by jharrington22 — CHANGES_REQUESTED** [review]

File: pkg/controller/customdomain/customdomain_controller.go [hunk 3]
```diff
@@ -24,10 +25,14 @@ import (
...
 var log = logf.Log.WithName("controller_customdomain")
 
+// managedIngressNames contains an array of known managed ingresscontroller
+var managedIngressNames = []string{"default", "apps2"}
+
 const (
 	ingressNamespace         = "openshift-ingress"
 	ingressOperatorNamespace = "openshift-ingress-operator"
 	dnsConfigName            = "cluster"
+	ingressLabelName         = "osd-custom-domains-owner"
```
[the remark this answers is not in the document]
Comment (dustman9000) [reply pkg/controller/customdomain/customdomain_controller.go:35]: Sure, but there's no built-in way to get the API name. We still need to define a constant: https://github.com/openshift/hive/blob/b8e8e2ba9b471445b9c76dc0798a456a4fa1d146/pkg/constants/constants.go#L127

File: pkg/controller/customdomain/customdomain_utils.go (hunk 1 above)
Comment (dustman9000) [reply pkg/controller/customdomain/customdomain_utils.go:120]: I've updated to return err if it's not the NotFound error.

File: pkg/controller/customdomain/customdomain_utils.go (hunk 2 above)
Comment (dustman9000) [reply pkg/controller/customdomain/customdomain_utils.go:144]: Updated to return err if it's not the NotFound error.

**Review by jharrington22 — APPROVED** [review]
/lgtm

**Comment by 2uasimojo** [comment]
/retest

OMG infra OOM

**Pull request merged**